jerry-core/ecma/operations/ecma-objects.c

```diff
@@ -1701,23 +1609,34 @@ ecma_op_object_define_own_property (ecma_object_t *obj_p, /**< the object */
 
         ecma_number_t num = ecma_string_to_number (property_name_p);
         ecma_string_t *num_to_str = ecma_new_ecma_string_from_number (num);
+        bool is_equal = ecma_compare_ecma_strings (property_name_p, num_to_str);
+        ecma_deref_ecma_string (num_to_str);
 
-        if (ecma_compare_ecma_strings (property_name_p, num_to_str))
+        if (is_equal)
         {
-          ecma_deref_ecma_string (num_to_str);
-
           return ecma_reject (property_desc_p->flags & ECMA_PROP_IS_THROW);
         }
 
         ecma_deref_ecma_string (num_to_str);
```

Comment:
Do we need the deref here? There is already one above.

Reply:
Good catch! Fixed.